src/dmd/backend/var.d

```diff
@@ -296,114 +296,175 @@ __gshared uint[256] tytab =
+
+        TYsptr     : "__ss *",
+        TYcptr     : "__cs *",
+        TYf16ptr   : "__far16 *",
+        TYfptr     : "__far *",
+        TYhptr     : "__huge *",
+        TYvptr     : "__handle *",
+        TYimmutPtr : "__immutable *",
+        TYsharePtr : "__shared *",
+        TYrestrictPtr : "__restrict *",
+        TYfgPtr    : "__fg *",
+        TYffunc    : "far C func",
+        TYfpfunc   : "far Pascal func",
+        TYfsfunc   : "far std func",
+        TYf16func  : "_far16 Pascal func",
+        TYnsysfunc : "sys func",
+        TYfsysfunc : "far sys func",
+        TYfref     : "__far &",
+
+        TYifunc    : "interrupt func",
+        TYmemptr   : "memptr",
+        TYident    : "ident",
+        TYtemplate : "template",
+        TYvtshape  : "vtshape",
+    ];
+
+    version(SCPP)
+    {
+        ret[TYschar] = "signed char";
+        ret[TYuchar] = "unsigned char";
+        ret[TYwchar_t] = "wchar_t";
+
+        ret[TYnullptr] = "nullptr_t";
+
+        ret[TYushort] = "unsigned short";
+        ret[TYuint] = "unsigned int";
+        ret[TYulong] = "unsigned long";
+
+        ret[TYldouble] = "long double";
+
+        ret[TYifloat] = "imaginary float";
+        ret[TYidouble] = "imaginary double";
+        ret[TYildouble] = "imaginary long double";
+
+        ret[TYcfloat] = "complex float";
+        ret[TYcdouble] = "complex double";
+        ret[TYcldouble] = "complex long double";
+
+        ret[TYllong] = "long long";
+        ret[TYullong] = "unsigned long long";
+
+        ret[TYschar16] = "signed char[16]";
+        ret[TYuchar16] = "unsigned char[16]";
+        ret[TYushort8] = "unsigned short[8]";
+        ret[TYulong4] = "unsigned long[4]";
+        ret[TYllong2] = "long long[2]";
+        ret[TYullong2] = "unsigned long long[2]";
+
+        ret[TYschar32] = "signed char[32]";
+        ret[TYuchar32] = "unsigned char[32]";
+        ret[TYushort16] = "unsigned short[16]";
+        ret[TYulong8] = "unsigned long[8]";
+        ret[TYllong4] = "long long[4]";
+        ret[TYullong4] = "unsigned long long[4]";
+
+        ret[TYschar64] = "signed char[64]";
+        ret[TYuchar64] = "unsigned char[64]";
+        ret[TYushort32] = "unsigned short[32]";
+        ret[TYulong16] = "unsigned long[16]";
+        ret[TYllong8] = "long long[8]";
+        ret[TYullong8] = "unsigned long long[8]";
+    } else version(MARS)
+    {
+        ret[TYschar] = "byte";
+        ret[TYuchar] = "ubyte";
+        ret[TYwchar_t] = "wchar";
+
+        ret[TYnullptr] = "typeof(null)";
+
+        ret[TYushort] = "ushort";
+        ret[TYuint] = "uint";
+        ret[TYulong] = "ulong";
+
+        ret[TYldouble] = "real";
+
+        ret[TYifloat] = "ifloat";
+        ret[TYidouble] = "idouble";
+        ret[TYildouble] = "ireal";
+
+        ret[TYcfloat] = "cfloat";
+        ret[TYcdouble] = "cdouble";
+        ret[TYcldouble] = "creal";
+
+        ret[TYullong] = ret[TYulong]; // c_ulong
+        ret[TYllong] = ret[TYlong]; // c_long
```

Comment:
I'm not sure about the naming for these types. They appear to be unused in D, but this corresponds to the c_ulong and c_long definitions. I have three different options here:
1. `TYullong` is the same as `TYulong` and therefore name it `ulong` (same approach for signed type -- `long`)
2. Leave unchanged (with the C naming)
3. Rename it to `c_long` and `c_ulong`.

Since `c_long` and `c_ulong` is an implementation defined druntime definition, I wouldn't consider the third option.